### PR TITLE
DOC: to_datetime format argument examples

### LINKED ISSUE #16669: DOC: new mini-section on specifying formats on datetime parsing
We have a single note on how to use ``format=`` kw in ``timeseries.rst``: https://github.com/pandas-dev/pandas/blob/master/doc/source/timeseries.rst#L178

I think let's make this a min-sub-section and provide links (or *maybe* just replicate) [python strptime docs](https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior) and show how to use the codings for specifying formats in some examples.

### PR DESCRIPTION
**Pull request by philipphanemann**
- [ ] closes #16669

### CODE REVIEW

File: doc/source/timeseries.rst [hunk 1]
```diff
@@ -191,6 +187,25 @@ or ``format``, use ``to_datetime`` if these are required.
 
     pd.Timestamp('2010/11/12')
 
+Providing a Format Argument
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+A specific ``format`` argument can be passed in addition to the datetime string.
```
Comment (jorisvandenbossche) [inline doc/source/timeseries.rst]: I would add something like "to ensure a specific and consistent parsing of the string"

File: doc/source/timeseries.rst [hunk 2]
```diff
@@ -191,6 +187,25 @@ or ``format``, use ``to_datetime`` if these are required.
...
+It will potentially speed up the conversion considerably and on versions later
+then 0.13.0.
```
Comment (jorisvandenbossche) [inline doc/source/timeseries.rst]: The "and on versionse later than 0.13.0" can be removed as well

File: doc/source/timeseries.rst (hunk 1 above)
Comment (jorisvandenbossche) [reply doc/source/timeseries.rst]: Can you also add a blankline between the title and the first sentence ?

File: doc/source/timeseries.rst [hunk 3]
```diff
@@ -175,12 +175,8 @@ you can pass the ``dayfirst`` flag:
    can't be parsed with the day being first it will be parsed as if
    ``dayfirst`` were False.
 
-.. note::
-   Specifying a ``format`` argument will potentially speed up the conversion
-   considerably and on versions later then 0.13.0 explicitly specifying
-   a format string of '%Y%m%d' takes a faster path still.
```
Comment (jorisvandenbossche) [inline doc/source/timeseries.rst]: @jreback we removed this part, as it is not a very clear sentence, and also does not seem to be true (https://github.com/pandas-dev/pandas/issues/17410)

**Review by jorisvandenbossche — APPROVED** [review]

File: doc/source/timeseries.rst [hunk 4]
```diff
@@ -191,6 +187,25 @@ or ``format``, use ``to_datetime`` if these are required.
...
+
+For example:
+
+.. ipython:: python
+
+    pd.to_datetime('2010/11/12', format='%Y/%m/%d')
+
+    pd.to_datetime('12-11-2010 00:00', format='%d-%m-%Y %H:%M')
+
+For further format options see: https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior
```
Comment (jreback) [inline doc/source/timeseries.rst]: can you format the link

`` `text-to-display <linktext>`__ ``

File: doc/source/timeseries.rst [hunk 5]
```diff
@@ -191,6 +187,25 @@ or ``format``, use ``to_datetime`` if these are required.
 
     pd.Timestamp('2010/11/12')
```
Comment (jreback) [inline doc/source/timeseries.rst]: can you add a section ref tag here as well.

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. small changes. pls rebase, ping on green.

File: doc/source/timeseries.rst (hunk 4 above)
Comment (philipphanemann) [reply doc/source/timeseries.rst]: I have put the linktext whithin the arrows and pushed. Hope that'll do

File: doc/source/timeseries.rst (hunk 5 above)
Comment (philipphanemann) [reply doc/source/timeseries.rst]: I have put a reference to the `pd.Timestamp` documentation as above and pushed.

File: doc/source/timeseries.rst (hunk 4 above)
Comment (philipphanemann) [reply doc/source/timeseries.rst]: After going through the contribution docs. I rebased and pushed the branch. Sorry for the inconvenience, that I first pushed to master.

**Comment by jorisvandenbossche** [comment]
@philipphanemann Thanks! 
(fixed the merge conflict and removed the link to "timestamp vs timespan/period" (I don't think this was a correct link)

**Pull request merged**